src/Symfony/Component/DependencyInjection/Compiler/CheckTypeDeclarationsPass.php

```diff
@@ -172,8 +175,12 @@ private function checkType(Definition $checkedDefinition, $value, \ReflectionPar
 
         if ($value instanceof Parameter) {
             $value = $this->container->getParameter($value);
-        } elseif (\is_string($value) && '%' === ($value[0] ?? '') && preg_match('/^%([^%]+)%$/', $value, $match)) {
-            $value = $this->container->getParameter($match[1]);
+        } elseif (\is_string($value)) {
+            if ('%' === ($value[0] ?? '') && preg_match('/^%([^%]+)%$/', $value, $match)) {
+                $value = $this->container->getParameter($match[1]);
+            } elseif ($envPlaceholderUniquePrefix && 0 === strpos($value, $envPlaceholderUniquePrefix)) {
+                $value = $this->container->resolveEnvPlaceholders($value, true);
```

Comment:
this means the command will fail if an env var is missing - not what I would expect from a linter
what about skipping these instead?

Reply:
Yes that will fail and that's fine to me. A missing env var is a problem to validate the container. Maybe we need a strict and lax mode.

Reply:
That's not fine to me: linting should be possible in build environments, that don't have env vars available.

Reply:
What about skipping them by default, but add a "--strict" option then? Resolving env vars is a legitimate case.